PublicMapping/districtbuilder-classic
# Make email configurable #155375259

## Overview

Makes email configurable so that sending email works.

### Checklist


## Testing Instructions

 * Add `DEBUG=True` to your `.env` file.
 * `./scripts/server`
 * `docker-compose exec django bash` in VM
 * `./manage.py shell`
 * Run this:
```
from django.core.mail import send_mail

send_mail(
    'Subject here',
    'Here is the message!.',
    'from@example.com',
    ['you@azavea.com'],
    fail_silently=False,
)
```
 * You should see the content of the email logged to stdout (useful for development) but no email was actually sent
 * Remove `DEBUG=True` from `.env`
 * Go here to generate an app password for your Google account (we're going to use Google's SMTP server to test actually sending email): https://myaccount.google.com/apppasswords
 * Make your `.env` file look like this:
```
MAILER_HOST=smtp.gmail.com
MAILER_PORT=587
MAILER_USER=<you@azavea.com>
MAILER_PASSWORD=<my_app_password>
MAILER_USE_TLS_OR_SSL=tls
```
 * Get into the Django shell again and run the same code as above, substituting 'you@azavea.com' with your email. You should get an email sent to that address.
 * Now make your `.env` file look like this to test SSL (note the port has changed as well):
```
MAILER_HOST=smtp.gmail.com
MAILER_PORT=465
MAILER_USER=<you@azavea.com>
MAILER_PASSWORD=<my_app_password>
MAILER_USE_TLS_OR_SSL=ssl
```
 * Run the code again from a shell and again you should receive an email to that address.
 * Test out the forgot password functionality by clicking "Forgot your password?" on the landing page and then putting your email in the bottom text input and clicking the "Send" button. You should receive an email at the address registered to that user. If needed, you can always create a new user with your email address, or log in with an existing user and change your email address via "My Account" in the top right.
 * Log in with your user to test the other known email sending feature: plan uploading. After logging in, click the "Upload Plan" tab. Select a zip file to upload (you can select a plan and request an index file to get a zip you can use). Select a body, give it a name, and upload plan. You should see a success modal and get an email about your upload.

Closes #155375259
